x/stake/genesis.go

```diff
@@ -32,11 +32,16 @@ func InitGenesis(ctx sdk.Context, k Keeper, data GenesisState) {
 	k.setPool(ctx, data.Pool)
 	k.setNewParams(ctx, data.Params)
 	for _, validator := range data.Validators {
+		// Staking assumes bonded validators are already stored, need to force update
+		validator.PoolShares.Status = sdk.Unbonded
+		k.setValidator(ctx, validator)
```

Comment:
Also if we do an `updateValidator` - as what is done in the next line - then we do not need to do a `setValidator` (done within `updateValidator`)